AsyncDisplayKit/Details/ASDataController.mm

```diff
@@ -112,7 +144,7 @@ - (void)_layoutNodes:(NSArray *)nodes atIndexPaths:(NSArray *)indexPaths withAni
     //TODO: There should be a fast-path that avoids all of this object creation.
     NSMutableArray *nodeBoundSizes = [[NSMutableArray alloc] initWithCapacity:kASDataControllerSizingCountPerProcessor];
     [subIndexPaths enumerateObjectsUsingBlock:^(NSIndexPath *indexPath, NSUInteger idx, BOOL *stop) {
-      ASSizeRange constrainedSize = [_dataSource dataController:self constrainedSizeForNodeAtIndexPath:indexPath];
+      ASSizeRange constrainedSize = constraintedSizeBlock(indexPath);
```

Comment:
Thanks for the feedback. I updated the diff to reflect a subclassing approach that I think will work much better with the addition of node kinds. @Adlai-Holler's addition this morning is a nice cherry on top of this whole thing.